### PR TITLE
[aptos-framework] introduce HashValue to aptos-stdlib and add block id to NewBlockEvent

### PR DESCRIPTION
**Pull request by zekun000**
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3147)

### CODE REVIEW

File: aptos-move/framework/aptos-stdlib/sources/hash.move [hunk 1]
```diff
@@ -14,4 +14,17 @@ module aptos_std::aptos_hash {
         // TODO: temporary mockup.
         pragma opaque;
     }
+
+    struct HashValue has copy, drop, store{
```
Comment (areshand) [inline aptos-move/framework/aptos-stdlib/sources/hash.move:18]: use it for table handle as well?

Comment (zekun000) [reply aptos-move/framework/aptos-stdlib/sources/hash.move:18]: that's the plan, I need to wait https://github.com/move-language/move/pull/391 landed and upgrade the dependency

File: aptos-move/framework/aptos-framework/sources/block.move [hunk 2]
```diff
@@ -103,8 +106,10 @@ module aptos_framework::block {
 
         let block_metadata_ref = borrow_global_mut<BlockResource>(@aptos_framework);
         block_metadata_ref.height = event::counter(&block_metadata_ref.new_block_events);
+        let id = aptos_hash::new_hash_value(*vector::borrow(&id, 0), *vector::borrow(&id, 1));
```
Comment (alinush) [inline aptos-move/framework/aptos-framework/sources/block.move:109]: Did you mean to use the same `id` variable name here?

**Review by alinush — COMMENTED** [review]
Thank you for moving the cryptographic hashes into `cryptography/`

File: crates/aptos-rest-client/src/lib.rs [hunk 3]
```diff
@@ -443,6 +452,7 @@ impl Client {
                 limit,
             )
             .await?;
+        println!("{:?}", response);
```
Comment (gregnazario) [inline crates/aptos-rest-client/src/lib.rs:455]: Leftover println

**Review by gregnazario — COMMENTED** [review]
I didn't see anything strange in it.

Are we going to be using this `HashValue` to replace all the other random hash values that we're currently storing?  We use `HashValue` for governance and other things

File: types/src/block_metadata.rs [hunk 4]
```diff
@@ -54,8 +55,12 @@ impl BlockMetadata {
     }
 
     pub fn get_prologue_move_args(self, signer: AccountAddress) -> Vec<MoveValue> {
+        let bytes = self.id.to_vec();
+        let low = u128::from_le_bytes(bytes[0..16].try_into().unwrap());
+        let high = u128::from_le_bytes(bytes[16..32].try_into().unwrap());
         vec![
             MoveValue::Signer(signer),
+            MoveValue::Vector(vec![MoveValue::U128(low), MoveValue::U128(high)]),
```
Comment (davidiw) [inline types/src/block_metadata.rs:63]: you don't want to give the illusion that this is just one value :(

Comment (davidiw) [reply types/src/block_metadata.rs:63]: how does this impact the various SDKs?

Comment (zekun000) [reply types/src/block_metadata.rs:63]: I don't know, updating one on rest client

**Review by davidiw — APPROVED** [review]

**Pull request merged**